src/provider/maestro.ts

```diff
@@ -0,0 +1,254 @@
+    const result = await fetch(
+      `${this.url}/accounts/${rewardAddress}`,
+      { headers: { "api-key": this.projectId, lucid } },
+    ).then((res) => res.json());
+    if (!result || result.message) {
+      return { poolId: null, rewards: 0n };
+    }
+    return {
+      poolId: result.delegated_pool || null,
+      rewards: BigInt(result.rewards_available),
+    };
+  }
+
+  async getDatum(datumHash: DatumHash): Promise<Datum> {
+    const result = await fetch(
+      `${this.url}/datum/${datumHash}`,
+      {
+        headers: { "api-key": this.projectId, lucid },
+      },
+    )
+      .then((res) => res.json())
+    if (!result || result.message) {
+      throw new Error(`No datum found for datum hash: ${datumHash}`);
+    }
+    return result.bytes;
+  }
+
+  awaitTx(txHash: TxHash, checkInterval = 3000): Promise<boolean> {
+    return new Promise((res) => {
+      const confirmation = setInterval(async () => {
+        const isConfirmed = await fetch(`${this.url}/transactions/${txHash}/cbor`, {
+          headers: { "api-key": this.projectId, lucid },
+        }).then((res) => res.json());
+        if (isConfirmed && !isConfirmed.message) {
+          clearInterval(confirmation);
+          await new Promise((res) => setTimeout(() => res(1), 1000));
+          return res(true);
+        }
+      }, checkInterval);
+    });
+  }
+
+  async submitTx(tx: Transaction): Promise<TxHash> {
+    const result = await fetch(`${this.url}/submit/tx`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/cbor",
+        "api-key": this.projectId,
+        lucid,
+      },
+      body: fromHex(tx),
+    }).then((res) => res.json());
+    if (!result || result.message) {
+      if (result?.code === 400) throw new Error(result.message);
+      else throw new Error("Could not submit transaction.");
+    }
+    return result;
+  }
+
+  private maestroUtxoToUtxo(result: MaestroUtxo): UTxO {
+    return {
+      txHash: result.tx_hash,
+      outputIndex: result.index,
+      assets: (() => {
+        const a: Assets = {};
+        result.assets.forEach((am) => {
+          a[am.unit.replace('#', '')] = BigInt(am.quantity);
+        });
+        return a;
+      })(),
+      address: result.address,
+      datumHash: result.datum ? result.datum.type == 'inline' ? undefined : result.datum.hash : undefined,
+      datum: result.datum ? result.datum.bytes : undefined,
+      scriptRef: result.reference_script ? result.reference_script.type == 'native' ? undefined : { type: result.reference_script.type == 'plutusv1' ? "PlutusV1" : "PlutusV2", script: applyDoubleCborEncoding(result.reference_script.bytes!) } : undefined
+    }
+  }
+}
+
+type MaestroDatumOptionType = "hash" | "inline";
+
+type MaestroDatumOption = {
+  type: MaestroDatumOptionType;
+  hash: string;
+  bytes?: string;  // Hex encoded datum CBOR bytes (`null` if datum type is `hash` and corresponding datum bytes have not been seen on-chain).
+  json?: object;
```

Comment:
The linter doesn't like `object` as type. Instead I would import the `Json` type from Lucid and use it here.